shared/js/event_dispatcher.js

```diff
@@ -199,7 +215,7 @@
             'Object to mix into already has "' + method + '" property defined!'
           );
         }
-        target[method] = eventDispatcher[method].bind(this);
+        this[method] = target[method] = eventDispatcher[method].bind(this);
```

Comment:
Ha-ha, I prefer mine just because I don't like to deal with ```call/apply``` if I can avoid that :) But it doesn't matter for me in the end, so will use your way :)